### PR TITLE
Reporting for memory_profiler

### PR DESCRIPTION
**Pull request by sagaru**
- Generation of report from memory_profiler
- Added reporting feature via logging module

### CODE REVIEW

**Comment by sagaru** [comment]
I wanted reporting the console output to a file, I found out that it can be done by passing a file io stream for the decorator.

Redirecting to a file was nice, but write will be continuous as long as process runs, so one has to use some FileHandler so that a the handler writes to other file, once the file reaches a certain file size limit. 

RotatingFileHandler of logging module was perfect for this job but memory_profiler module was not supporting it. I wrote a small class to make it work.

Because of abundant entries, it was even hard to point out where the increment has happened, added an option to filter entries with only increments in report.

File: memory_profiler.py [hunk 1]
```diff
@@ -808,6 +809,25 @@ def inner_wrapper(f):
             return profile(f, stream=stream, precision=precision)
         return inner_wrapper
 
+class LogFile(object):
+    """File-like object to log text using the `logging` module and the log report can be customised."""
```
Comment (fabianp) [inline memory_profiler.py]: Could you please describe in the docstring the arguments name and reportIncrementFlag ?

File: README.rst [hunk 2]
```diff
@@ -227,6 +227,47 @@ args, kw)``. For example::
...
+Sending everything to the log file while running the memory_profiler
+could be cumbersome and one can choose only entries with increments
+by passing True to reportIncrementFlag, where reportIncrementFlag is
+a parameter to LogFile class of memory profiler module.
+
+    >>> from memory_profiler import LogFile
+    >>> import sys
+    >>> sys.stdout = LogFile('memory_profile_log', reportIncrementFlag=False)
+
+    For details refer: examples/reportingViaLogger.py
```
Comment (fabianp) [inline README.rst]: please use lowercase for filenames, i.e. rename this last one to reporting_logger.py

Comment (fabianp) [inline README.rst]: And add your name (and a link if you want too) to the end for the README with your contribution

**Comment by fabianp** [comment]
Great, its merged. Thanks a lot for this

**Pull request merged**